providers/pool.rb

```diff
@@ -232,6 +236,38 @@ def configure
       shell_out!(cmd)
       @new_resource.updated_by_last_action(true)
     end
+    if is_new_start_mode
+      was_updated = true
+      cmd = "#{appcmd(node)} set apppool \"/apppool.name:#{@new_resource.pool_name}\""
+      cmd << " /startMode:#{@new_resource.start_mode.to_s}"
+      Chef::Log.debug(cmd)
+      shell_out!(cmd)
+      @new_resource.updated_by_last_action(true)
```

Comment:
I don't get the point of this line (repeated for each block), combined with the `was_updated` variable which handles this behavior at the end of the `configure` method.